src/test/java/com/devonfw/sample/archunit/ThirdPartyRulesTest.java

```diff
@@ -1,31 +1,244 @@
+
+  @ArchTest
+  public static ArchRule check_object_dependency = noClasses().should().dependOnClassesThat()
+      .haveFullyQualifiedName("com.google.common.base.Objects")
+      .because("Use Java standards instead (java.util.Objects).");
+
+  @ArchTest
+  public static ArchRule check_converter_dependency = noClasses().should().dependOnClassesThat()
+      .haveFullyQualifiedName("javax.persistence.Convert")
+      .because("Use the javax.persistence.Converter annotation on a custom converter"
+          + " which implements the javax.persistence.AttributeConverter instead of the 'javax.persistance.Convert' annotation");
+
+  @ArchTest
+  public static ArchRule check_mysema_dependency = noClasses().should().dependOnClassesThat()
+      .resideInAPackage("com.mysema.query..")
+      .because("Use official QueryDSL (com.querydsl.* e.g. from com.querydsl:querydsl-jpa).");
+
+  /*
+   * E1
+   */
+  private static boolean isApiScopedClassUsingTransactional(JavaClass source, String targetPackageFullName) {
+
+    PackageStructure sourcePkg = PackageStructure.of(source);
+
+    if (sourcePkg.isScopeApi() && targetPackageFullName.equals("javax.transaction.Transactional")) {
+      return true;
+    }
+    return false;
+  }
+
+  private static boolean isUsingSpringframeworkTransactionalAnnotation(String targetPackageFullName) {
+
+    if (targetPackageFullName.equals("org.springframework.transaction.annotation.Transactional")) {
+      return true;
+    }
+    return false;
+  }
+
+  public static final ArchCondition<JavaClass> misuse_springframework_transactional_annotation = new ArchCondition<JavaClass>(
+      "misuse @Transactional (Rule-E1)") {
+    @Override
+    public void check(JavaClass source, ConditionEvents events) {
+
+      for (Dependency dependency : source.getDirectDependenciesFromSelf()) {
+        String targetFullName = dependency.getTargetClass().getFullName();
+        String targetClassDescription = dependency.getDescription();
+        if (isUsingSpringframeworkTransactionalAnnotation(targetFullName) == true) {
+          String message = String.format(
+              "Use JEE standard (javax.transaction.Transactional from javax.transaction:javax.transaction-api:1.2+). The use (%s) is discouraged.  Violated in (%s)",
+              targetFullName, targetClassDescription);
+          events.add(new SimpleConditionEvent(source, true, message));
+        }
+        /*
+         * In case the project has a classic architecture using scopes, check that no API scoped class is using
+         * 'javax.transaction.Transactional'
+         */
+        if (isApiScopedClassUsingTransactional(source, targetFullName) == true) {
+          String message = String.format(
+              "The use of @Transactional in API is discouraged. Instead use it to annotate implementations. Violated in (%s)",
+              targetClassDescription);
+          events.add(new SimpleConditionEvent(source, true, message));
+        }
```

Comment:
This is correct and shall remain but the rule name is `misuse_springframework_transactional_annotation` and this has nothing to do with that.

Reply:
Changed the remaining scope API condition name to `verifyingTransactionalAnnotationIsNotUsedInsideApi` and the rule name to`verifyingProperTransactionalUseFromJee`.